echopype/echodata/simrad.py

```diff
@@ -0,0 +1,233 @@
+        raise RuntimeError("encode_mode='power' not allowed when waveform_mode='BB'!")
+
+    # make sure that we have BB and complex inputs, if pulse compression is selected
+    if pulse_compression and ((waveform_mode != "BB") or (encode_mode != "complex")):
+        raise RuntimeError(
+            "Pulse compression can only be used with "
+            "waveform_mode='BB' and encode_mode='complex'"
+        )
+
+
+def _check_mode_input_with_data_EK60(
+    echodata: EchoData, waveform_mode: str, encode_mode: str
+) -> Optional[str]:
+    """
+    Ensures that the provided ``waveform_mode`` and ``encode_mode`` are consistent
+    with the EK60-like data supplied by ``echodata``.
+
+    Parameters
+    ----------
+    echodata: EchoData
+        An ``EchoData`` object holding the data
+    waveform_mode : {"CW", "BB"}
+        Type of transmit waveform
+    encode_mode : {"complex", "power"}
+        Type of encoded return echo data
+
+    Returns
+    -------
+    power_ed_group: str, optional
+        The ``EchoData`` beam group path containing the power data
+    """
+
+    # initialize power EchoData group value
+    power_ed_group = None
+
+    # EK60-like sensors must have 'power' and 'CW' modes only
+    if waveform_mode != "CW":
+        raise RuntimeError("Incorrect waveform_mode input provided!")
+    if encode_mode != "power":
+        raise RuntimeError("Incorrect encode_mode input provided!")
+
+    # ensure that no complex data exists (this should never be triggered)
+    # TODO: is this necessary? I just wanted to provide a check against the data ...
+    if "backscatter_i" in echodata["Sonar/Beam_group1"].variables:
+        raise RuntimeError(
+            "Provided echodata object does not correspond to an EK60-like "
+            "sensor, but is labeled as data from an EK60-like sensor!"
+        )
+    else:
+        power_ed_group = "Sonar/Beam_group1"
+
+    return power_ed_group
+
+
+def _check_mode_input_with_data_EK80(
+    echodata: EchoData, waveform_mode: str, encode_mode: str
+) -> Tuple[Optional[str], Optional[str]]:
+    """
+    Ensures that the provided ``waveform_mode`` and ``encode_mode`` are consistent
+    with the EK80-like data supplied by ``echodata``.
+
+    Parameters
+    ----------
+    echodata: EchoData
+        An ``EchoData`` object holding the data
+    waveform_mode : {"CW", "BB"}
+        Type of transmit waveform
+    encode_mode : {"complex", "power"}
+        Type of encoded return echo data
+
+    Returns
+    -------
+    power_ed_group: str, optional
+        The ``EchoData`` beam group path containing the power data
+    complex_ed_group: str, optional
+        The ``EchoData`` beam group path containing the complex data
+    """
+
+    # initialize power and complex EchoData group values
+    power_ed_group = None
+    complex_ed_group = None
+
+    # EK80-like sensors
+    if echodata.sonar_model in ["EK80", "ES80", "EA640"]:
```

Comment:
seems that this check can be removed since this function is only called when sonar_model is one of those listed, under `check_waveform_encode_mode`

Reply:
You are correct, I will go ahead and remove this check.